packages/teams-js/src/public/media.ts

```diff
@@ -408,34 +412,69 @@ export namespace media {
      * @hidden
      * Hide from docs
      * --------
+     *
+     * Function to notify the host client to programatically control the experience
+     * @param mediaEvent indicates what the event that needs to be signaled to the host client
+     * @returns A promise resolved promise
+     */
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent): Promise<void>;
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     *
+     * @deprecated
+     * As of 2.0.0-beta.3, please use {@link media.MediaController.notifyEventToHost media.MediaController.notifyEventToHost(mediaEvent: MediaControllerEvent): Promise\<void\>} instead.
+     *
      * Function to notify the host client to programatically control the experience
      * @param mediaEvent indicates what the event that needs to be signaled to the host client
      * Optional; @param callback is used to send app if host client has successfully handled the notification event or not
      */
-    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void {
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): void;
+    protected notifyEventToHost(mediaEvent: MediaControllerEvent, callback?: (err?: SdkError) => void): Promise<void> {
       ensureInitialized(FrameContexts.content, FrameContexts.task);
-      const err = isApiSupportedOnMobile(nonFullScreenVideoModeAPISupportVersion);
-      if (err) {
-        if (callback) {
-          callback(err);
-        }
-        return;
-      }
 
-      const params: MediaControllerParam = { mediaType: this.getMediaType(), mediaControllerEvent: mediaEvent };
-      sendMessageToParent('media.controller', [params], (err?: SdkError) => {
-        if (callback) {
-          callback(err);
-        }
-      });
+      return Promise.resolve()
+        .then(() => {
+          return throwExceptionIfMobileApiIsNotSupported(nonFullScreenVideoModeAPISupportVersion);
```

Comment:
This revision looks much better :) 
You don't need the `return` before the `throwExceptionIfMobileApiIsNotSupported`, it will go to catch block anyway.

Bonus Tip: As people have commented below, there the utility functions which hide the ugly code if callback stuff. I would suggest if you can use any existing utility functions (eg:`callCallbackWithSdkErrorFromPromiseAndReturnPromise`), your code will be much cleaner. Else you can write up your own utility function, so it will help incase same scenarios pop up.

Reply:
Agreed, this revision looks really good.

(this is another return that we don't technically need, but it isn't hurting anything)